inference-engine/thirdparty/clDNN/src/gpu/non_max_suppression_cpu.cpp

```diff
@@ -387,10 +387,13 @@ struct non_max_suppression_cpu : typed_primitive_impl<non_max_suppression> {
         }
 
         auto& stream = instance.get_network().get_stream();
+        auto ev = stream.create_user_event(false);
 
         run(instance);
 
-        return stream.create_user_event(true);
+        ev->wait();
```

Comment:
@vladimir-paramuzov  Why is it `wait` instead of `set`? FYI @sungeunk

Reply:
@isanghao good catch. This was a mistake which could lead to app hang. Fixed. Also, changed behavior of wait_impl for user event to throw an exception if wait() is called before set()

Reply:
It is weird that it was not catched from dlbenchmark... I checked with the old commit for face-detection-0206, and it hanged.